src/components/calcite-scrim/calcite-scrim.tsx

```diff
@@ -0,0 +1,52 @@
+import { Component, Host, Prop, h } from "@stencil/core";
```

Comment:
fixme: Does this need to be a component? I'm wondering if we want this to be a public component that people could use?

It might make more sense as a functional component.

Reply:
@driskull I'll defer to your judgement. Dunno if it makes a difference, but this is likely to be transferred to calcite-components before long.

Reply:
I'm just thinking this might be a candidate for a functional component because it wouldn't be used on its own. It would always be used within another component, right? Also, we don't want a user to use this component right?

Reply:
If it's a FunctionalComponent, will I need to add its styles to **_base.scss**?

Reply:
I think may also need to have the states `loading` and `disabled`.

Reply:
Probably a mixin that components could share/import. Let's talk about this component before we decide to make it a functional component. Maybe a quick meeting or wait till friday?

Reply:
I'll be in training on Friday.

Reply:
I definitely support making this stuff more DRY. 💯 
I agree that we should explore other ways to abstract this code. A functional component may be best. Mixins might also work.

Reply:
> If it's a FunctionalComponent, will I need to add its styles to _base.scss?

Yea you'll need to add either a mixin or a class to the global styles for it to be available across components. Double-checked and it doesn't seem a FunctionalComponent supports having it's own CSS, unless you inline it.